### PR TITLE
Parse JSON menu_item for SQLite and skip blank items

### PR DESCRIPTION
**Pull request by pchemguy**
Use parse_json helper to enable the use of the object form of "shell" with JSON containing menu items. Also in this pull request is an explicit filter on blank items (for example, generated via `NULL AS menu_item` or via `'' AS menu_item`). The latter feature simplifier creation of dynamic menus based on certain conditions, such as whether the user is authenticated or not. 

Note, however, `'' AS menu_item` presently fails due to parse_json and should not be used. `'{}' AS menu-item` apparently works fine.

### CODE REVIEW

File: sqlpage/templates/shell.handlebars [hunk 1]
```diff
@@ -73,28 +73,36 @@
...
-                                    {{#each this.submenu}}
-                                        <a class="dropdown-item" href="{{this.link}}">
-                                            {{this.title}}
-                                        </a>
-                                    {{/each}}
-                                </div>
-                            {{/if}}
-                            </li>
+                        {{#if (eq (typeof (parse_json this)) 'object')}}
+                            {{#with (parse_json this)}}
```
Comment (lovasoa) [inline sqlpage/templates/shell.handlebars:77]: This parses the same json string twice

**Review by lovasoa — CHANGES_REQUESTED** [review]

**Comment by pchemguy** [comment]
There was another issue. `'Help' AS menu_item` would be passed to parse_json causing an unhandled error. So I had to create a more elaborate check removing the first parse_json and only keeping it in the "with" block where a preliminary check is passed.

**Comment by lovasoa** [comment]
This is super useful ! 

Can we add an example to the documentation for these "conditional" menu items ? Otherwise people would never know the possibility even exists.

**Comment by pchemguy** [comment]
I have added a basic example. A more elaborate example will be a part of separate pull request.

**Pull request merged**